src/scripts/main.js

```diff
@@ -1,11 +1,42 @@
+/* eslint-disable max-len */
+/* eslint-disable no-template-curly-in-string */
 'use strict';
 
+const body = document.querySelector('body');
+
 function waitFor(element, eventName) {
-  // write your code here
+  const resolveMessage = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+  return new Promise((resolve) => {
+    switch (eventName) {
+      case 'click':
+        return element.addEventListener('click', () => {
+          resolve(resolveMessage);
+        });
+
+      case 'contextmenu':
+        return element.addEventListener('contextmenu', () => {
+          resolve(resolveMessage);
+        });
+
+      case 'blur':
+        return element.addEventListener('blur', () => {
+          resolve(resolveMessage);
+        });
+
+      default:
+        return 0;
```

Comment:
The `default` case in the `switch` statement returns `0`, which might not be the best practice. Consider throwing an error or returning a rejected promise to handle unsupported events more gracefully.